src/Handler/StreamHandler.php

```diff
@@ -106,18 +107,30 @@ private function checkDecode(array $options, array $headers, $stream)
     {
         // Automatically decode responses when instructed.
         if (!empty($options['decode_content'])) {
-            foreach ($headers as $key => $value) {
-                if (strtolower($key) == 'content-encoding') {
-                    if ($value[0] == 'gzip' || $value[0] == 'deflate') {
-                        return new Psr7\InflateStream(
-                            Psr7\stream_for($stream)
-                        );
+            $normalizedKeys = array_combine(array_map('strtolower', array_keys($headers)), array_keys($headers));
```

Comment:
I'm not sure if I understand you. Your recommendation is to change this line to:
```php
$normalizedKeys = array_combine(array_keys(array_change_key_case($headers))), array_keys($headers));
```
or to change it to:
```php
$lcheaders = array_change_key_case($headers));
```
and then, return the `$lcheaders` array overwriting the original headers case?